qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateDataSupport.java

```diff
@@ -0,0 +1,161 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IAnnotatable;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.qute.commons.datamodel.resolvers.ValueResolverInfo;
+import com.redhat.qute.jdt.QuteSupportForTemplate;
+import com.redhat.qute.jdt.internal.resolver.ITypeResolver;
+import com.redhat.qute.jdt.template.datamodel.AbstractAnnotationTypeReferenceDataModelProvider;
+import com.redhat.qute.jdt.template.datamodel.SearchContext;
+import com.redhat.qute.jdt.utils.AnnotationUtils;
+
+/**
+ * Template data support.
+ *
+ * <code>
+ * &#64;TemplateData
+ * class Item {
+ *
+ * 		public final BigDecimal price;
+ *
+ * 		public Item(BigDecimal price) {
+ * 			this.price = price;
+ * 		}
+ *
+ * 		public BigDecimal getDiscountedPrice() {
+ * 			return price.multiply(new BigDecimal("0.9"));
+ * 		}
+ * }
+ * </code>
+ *
+ *
+ * @see https://quarkus.io/guides/qute-reference#template_data
+ *
+ */
+public class TemplateDataSupport extends AbstractAnnotationTypeReferenceDataModelProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateDataSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+			TEMPLATE_DATA_ANNOTATION
+
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+			SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (!(javaElement instanceof IAnnotatable)) {
+			return;
+		}
+		IAnnotation templateData = AnnotationUtils.getAnnotation((IAnnotatable) javaElement, TEMPLATE_DATA_ANNOTATION);
+		if (templateData == null) {
+			return;
+		}
+		if (javaElement instanceof IType) {
+			IType type = (IType) javaElement;
+			collectResolversForTemplateData(type, templateData, context.getDataModelProject().getValueResolvers(),
+					monitor);
+		}
+	}
+
+	private static void collectResolversForTemplateData(IType type, IAnnotation templateData,
+			List<ValueResolverInfo> resolvers, IProgressMonitor monitor) {
+		try {
+			ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(type);
+			IField[] fields = type.getFields();
+			for (IField field : fields) {
+				// May want to check method access? i.e. only resolve public
+				IAnnotation fieldTemplateData = AnnotationUtils.getAnnotation(field, TEMPLATE_DATA_ANNOTATION);
+				collectResolversForTemplateData(field, fieldTemplateData != null ? fieldTemplateData : templateData,
+						resolvers, typeResolver);
+			}
+			IMethod[] methods = type.getMethods();
+			for (IMethod method : methods) {
+				// May want to check method access? i.e. only resolve public

```

Comment:
You need to check if it is public and static method.

`if (Flags.isPublic(method.getFlags()) && Flags.isStatic(method.getFlags()))`